awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsSpanMetricsProcessor.java

```diff
@@ -117,8 +122,10 @@ public void onEnd(ReadableSpan span) {
     Map<String, Attributes> attributeMap =
         generator.generateMetricAttributeMapFromSpan(spanData, resource);
 
-    for (Map.Entry<String, Attributes> attribute : attributeMap.entrySet()) {
-      recordMetrics(span, spanData, attribute.getValue());
+    if (!isEc2MetadataSpan(attributeMap)) {
+      for (Map.Entry<String, Attributes> attribute : attributeMap.entrySet()) {
+        recordMetrics(span, spanData, attribute.getValue());
+      }
```

Comment:
Could you move this check inside `recordMetrics` under the existing check there? 

```
// Only record metrics if non-empty attributes are returned.
    if (!attributes.isEmpty()) {
      recordErrorOrFault(spanData, attributes);
      recordLatency(span, attributes);
    }
```